Source/JavaScriptCore/wasm/WasmBBQJIT.cpp

```diff
@@ -3571,6 +3593,25 @@ PartialResult WARN_UNUSED_RETURN BBQJIT::addThrow(unsigned exceptionIndex, Argum
     return { };
 }
 
+PartialResult WARN_UNUSED_RETURN BBQJIT::addThrowRef(Value exception, Stack&)
+{
+
+    LOG_INSTRUCTION("ThrowRef", exception);
+
+    emitMove(exception, Location::fromGPR(GPRInfo::argumentGPR1));
+    consume(exception);
+
+    ++m_callSiteIndex;
+    bool mayHaveExceptionHandlers = !m_hasExceptionHandlers || m_hasExceptionHandlers.value();
+    if (mayHaveExceptionHandlers) {
+        m_jit.store32(CCallHelpers::TrustedImm32(m_callSiteIndex), CCallHelpers::tagFor(CallFrameSlot::argumentCountIncludingThis));
+        flushRegisters();
+    }
+    m_jit.move(GPRInfo::wasmContextInstancePointer, GPRInfo::argumentGPR0);
+    emitRethrowImpl(m_jit);
```

Comment:
Why is it `rethrow`? If it needs to be `rethrow`, I wonder if we should rename `emitRethrowImpl` and operations to `emitThrowRefImpl` etc. and implement rethrow with them.

Reply:
These two end up sharing the same path once we find the exception pointer to throw, I think it'd make sense to rename it though since this is the "main" exception spec now.